packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenPayment.sol

```diff
@@ -0,0 +1,286 @@
+    function setAirnodeToMinimumWhitelistDuration(
+        uint64 _minimumWhitelistDuration
+    ) external override {
+        require(
+            hasAirnodeToMinimumWhitelistDurationSetterRoleOrIsManager(
+                msg.sender
+            ),
+            "Not Airnode to minimum whitelist duration setter"
+        );
+        require(
+            _minimumWhitelistDuration != 0 &&
+                _minimumWhitelistDuration >= DEFAULT_MINIMUM_WHITELIST_DURATION,
+            "Invalid duration"
+        );
+        airnodeToMinimumWhitelistDuration[
+            msg.sender
+        ] = _minimumWhitelistDuration;
+        emit SetAirnodeToMinimumWhitelistDuration(
+            _minimumWhitelistDuration,
+            msg.sender
+        );
+    }
+
+    /// @notice Called by an Airnode to payment destination setter to set the
+    /// address to which payments will be transferred
+    /// @param _paymentDestination Payment destination address
+    function setAirnodeToPaymentDestination(address _paymentDestination)
+        external
+        override
+    {
+        require(
+            hasAirnodeToPaymentDestinationSetterRoleOrIsManager(msg.sender),
+            "Not Airnode to payment destination setter"
+        );
+        require(
+            _paymentDestination != address(0),
+            "Invalid destination address"
+        );
+        airnodeToPaymentDestination[msg.sender] = _paymentDestination;
+        emit SetAirnodeToPaymentDestination(_paymentDestination, msg.sender);
+    }
+
+    /// @notice Make payments to gain access to Airnode
+    /// @dev In order for this function to be able to extend the whitelisting,
+    /// the Airnode operator must first grant the whitelist expiration extender
+    /// role to this contract, otherwise it will revert.
+    /// chainId-airnode-endpoint-requester pair gets authorized for the
+    /// duration specified. The amount to be paid is determined by the fee set
+    /// in the AirnodeFeeRegistry contract and the token price.
+    /// @param _chainId Id of the chain
+    /// @param _airnode Airnode address
+    /// @param _endpointId Id of the endpoint
+    /// @param _requesterAddress Requester address
+    /// @param _whitelistDuration Duration in seconds for which the requester
+    /// will be whitelisted
+    function makePayment(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        uint64 _whitelistDuration
+    ) external override {
+        require(_chainId != 0, "Invalid chainId");
+        require(_airnode != address(0), "Invalid Airnode address");
+        require(_requesterAddress != address(0), "Invalid requester address");
+        require(
+            _whitelistDuration != 0 &&
+                _whitelistDuration <= DEFAULT_MAXIMUM_WHITELIST_DURATION &&
+                _whitelistDuration >= getMinimumWhitelistDuration(_airnode),
+            "Invalid whitelist duration"
```

Comment:
At first I thought the same but I then I thought of the use case and realized that an Airnode operator would probably only want to set the minimum (instead of charging by day they might want to charge by month minimum, etc). I could add a setter to the maximum too but that would require a new role that might never be used.